### PR TITLE
feat(genesis): separating chain param from genesis param

### LINKED ISSUE #1341: Set fee paramaetrs to zero in genesis doc
Once creating a new genesis file for the LocalNet the deprecated fee parameters set to default values, like:
```
      "fee_fraction": 0.0001,
      "minimum_fee": 1000,
      "maximum_fee": 1000000, 
```

These parameters should set to zero and mark as deprecated.

### PR DESCRIPTION
**Pull request by kehiy**
## Description

- Fixes #1341

### CODE REVIEW

File: tests/main_test.go [hunk 1]
```diff
@@ -39,11 +38,11 @@ var (
 )
 
 const (
-	tNodeIdx1      = 0
-	tNodeIdx2      = 1
-	tNodeIdx3      = 2
-	tNodeIdx4      = 3
-	tTotalNodes    = 4 // each node has 3 validators
+	tNodeIdx1 = iota
```
Comment (b00f) [inline tests/main_test.go:41]: It is not an enumerator. Please revert this.

Comment (kehiy) [reply tests/main_test.go]: Fixed.

File: sync/firewall/firewall_test.go [hunk 2]
```diff
@@ -354,7 +354,7 @@ func TestNetworkFlagsTestnet(t *testing.T) {
 
 func TestNetworkFlagsLocalnet(t *testing.T) {
 	td := setup(t, nil)
-	td.state.TestParams.BlockVersion = 0x3f // changing genesis hash
+	td.state.Genesis().Params().BlockVersion = 0x3f // changing genesis hash
```
Comment (b00f) [inline sync/firewall/firewall_test.go:357]: Why Genesis here?

File: state/state_test.go [hunk 3]
```diff
@@ -55,7 +54,7 @@ func setup(t *testing.T) *testData {
 
 	genTime := util.RoundNow(10).Add(-8640 * time.Second)
 
-	params := param.DefaultParams()
+	params := genesis.DefaultGenesisParams()
```
Comment (b00f) [inline state/state_test.go:57]: Please rename it to `genParams`

File: tests/main_test.go [hunk 4]
```diff
@@ -119,7 +118,7 @@ func TestMain(m *testing.M) {
 	vals[1] = validator.NewValidator(tValKeys[tNodeIdx2][0].PublicKey(), 1)
 	vals[2] = validator.NewValidator(tValKeys[tNodeIdx3][0].PublicKey(), 2)
 	vals[3] = validator.NewValidator(tValKeys[tNodeIdx4][0].PublicKey(), 3)
-	params := param.DefaultParams()
+	params := genesis.DefaultGenesisParams()
```
Comment (b00f) [inline tests/main_test.go:121]: Please rename it to `genParams`

File: sync/firewall/firewall_test.go (hunk 2 above)
Comment (kehiy) [reply sync/firewall/firewall_test.go:357]: changing test params won't change the version and test will fail.

**Review by b00f — APPROVED** [review]

**Pull request merged**